src/server/common.cc

```diff
@@ -190,6 +193,13 @@ bool ParseDouble(string_view src, double* value) {
   return true;
 }
 
+void OpArgs::RecordJournal(string_view cmd, ArgSlice args) const {
+  auto journal = shard->journal();
+  CHECK(journal);
+  auto opcode = tx->IsMulti() ? journal::Op::MULTI_COMMAND : journal::Op::COMMAND;
+  journal->RecordEntry(tx->txid(), opcode, db_cntx.db_index, 1, make_pair(cmd, args));
```

Comment:
It should be like this, right?

Reply:
How about creating creating this function in Transaction class, that will get a payload and will be called from LogJournalOnShard and from the flows you want to change the journal. Instead of having OpArgs::RecordJournal funciton

Reply:
United them, I still kept the function though. Should I remove it too?